internal/server/openapi/api_model_registry_service_service.go

```diff
@@ -450,7 +452,16 @@ func (s *ModelRegistryServiceAPIService) UpdateInferenceService(ctx context.Cont
 		return Response(http.StatusBadRequest, model.Error{Message: err.Error()}), nil
 	}
 	entity.Id = &inferenceserviceId
-	result, err := s.coreApi.UpsertInferenceService(entity)
+	existing, err := s.coreApi.GetInferenceServiceById(inferenceserviceId)
+	if err != nil {
+		status := api.ErrToStatus(err)
+		return Response(status, model.Error{Message: err.Error()}), nil
+	}
+	update, err := s.reconciler.UpdateExistingInferenceService(converter.NewOpenapiUpdateWrapper(existing, entity))
+	if err != nil {
+		return Response(http.StatusBadRequest, model.Error{Message: err.Error()}), nil
+	}
+	result, err := s.coreApi.UpsertInferenceService(&update)
```

Comment:
pattern using the "reconciler" is applied to MR logical model entities:

1. get from MLMD the current entity value
2. apply the reconciler, that means applying the "patch" deltas to the existing value
3. upsert the "to-be" value back into MLMD, via the core layer upsert.

👍